WordPress/gutenberg
# Cover Block: Fix block deprecation when fixed background is enabled

Fixes #51473 

## What?
This PR fixes a problem where the block deprecation fails in cover blocks where "Fixed Background" is enabled.

https://github.com/WordPress/gutenberg/assets/54422211/33c9493c-9873-4450-83f3-d7345c8225b4

## Why?
#46969 adds `tagName` attributes to change the default `<div>` of the cover block to other elements. Because all but the newest markup has no default `tagName` attributes, the deprecation from `v1` to `v9` has an additional `migrate` property that defines a default `tagName` attributes. 

However, `v10` and `v11` deprecation appear to be missing that process. As a result, if the old markup matches a deprecation after `v10`, the save function will fail to process it because the `tagName` attribute is `undefined`.

## How?
I have added a similar migrate process to `v10` and `v11` deprecation.

## Testing Instructions

Insert a cover block using the following markup:

```html

<div class="wp-block-cover has-parallax" style="background-image:url(https://live.staticflickr.com/2328/2061718046_64bbbc0a04_b.jpg)">
	<span aria-hidden="true" class="wp-block-cover__background has-background-dim-70 has-background-dim"></span>
	<div class="wp-block-cover__inner-container">
		
		<p>Fixed background</p>
		
	</div>
</div>

```

When the focus is removed from the code editor, confirm that a migration is performed to ensure that the block is updated with the latest markup.

### Another test instructions

- In WordPress 6.0 where the cover block has this markup, insert the cover block and activate "Fixed Background". Please disable the Gutenberg plugin.
- Update WordPress to the latest version and activate the Gutenberg plugin.
- Confirm that the migration is properly executed when the block editor is opened.
